Index: websites/F/freeCodeCamp/presence.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>const presence = new Presence({\n  clientId: \"797749214175035412\"\n});\n\nconst timeStamp = Math.floor(Date.now() / 1000);\n\npresence.on(\"UpdateData\", () => {\n  const presenceData: PresenceData = {\n    largeImageKey: \"logo\",\n    startTimestamp: timeStamp\n  };\n  const page = document.location.pathname;\n  if (page.startsWith(\"/learn/\")) {\n    presenceData.details = \"Learning:\";\n    presenceData.state = \"Writing Code!\";\n  } else if (page.startsWith(\"/login\")) {\n    presenceData.details = \"Viewing Page:\";\n    presenceData.state = \"Signing In\";\n  } else if (page.startsWith(\"/news/\")) {\n    presenceData.details = \"Viewing Page:\";\n    presenceData.state = \"Reading the news\";\n  } else if (page.startsWith(\"/learn\")) {\n    presenceData.details = \"Deciding:\";\n    presenceData.state = \"Choosing what to learn today \uD83E\uDD14\";\n  }\n  presence.setActivity(presenceData);\n});\n
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/websites/F/freeCodeCamp/presence.ts b/websites/F/freeCodeCamp/presence.ts
--- a/websites/F/freeCodeCamp/presence.ts	(revision 9812e205c6f39a0c3cccc07e454395ffd88ec6b4)
+++ b/websites/F/freeCodeCamp/presence.ts	(date 1612299758500)
@@ -1,27 +1,41 @@
 const presence = new Presence({
   clientId: "797749214175035412"
-});
-
-const timeStamp = Math.floor(Date.now() / 1000);
+}),
+  timeStamp = Math.floor(Date.now() / 1000),
+  camps: {[index: string]:any} = {
+    "/basic-html-and-html5": "Basic HTML and HTML5 | Responsive Web Design",
+    "/basic-css": "Basic CSS | Responsive Web Design",
+    "/applied-visual-design": "Applied Visual Design | Responsive Web Design",
+    "/applied-accessibility": "Applied Accessibility | Responsive Web Design",
+    "/responsive-web-design-principles": "Responsive Web Design Principles | Responsive Web Design",
+    "/css-flexbox": "CSS Flexbox | Responsive Web Design"
+}
 
 presence.on("UpdateData", () => {
   const presenceData: PresenceData = {
     largeImageKey: "logo",
     startTimestamp: timeStamp
   };
-  const page = document.location.pathname;
-  if (page.startsWith("/learn/")) {
-    presenceData.details = "Learning:";
-    presenceData.state = "Writing Code!";
-  } else if (page.startsWith("/login")) {
-    presenceData.details = "Viewing Page:";
-    presenceData.state = "Signing In";
-  } else if (page.startsWith("/news/")) {
-    presenceData.details = "Viewing Page:";
-    presenceData.state = "Reading the news";
-  } else if (page.startsWith("/learn")) {
-    presenceData.details = "Deciding:";
-    presenceData.state = "Choosing what to learn today 🤔";
+  let page = document.location.pathname;
+  if (["/learn", "/learn/"].includes(page)) {
+    if(document.getElementById("javascript-algorithms-and-data-structures") != null) {
+      presenceData.details = "Deciding:";
+      presenceData.state = "Choosing what to learn today 🤔";
+    } else {
+      page = page.substring(page.lastIndexOf('/'))
+      if(page.endsWith("/")) {
+        page = page.replace(/[/]$/gm, "")
+        console.log(page);
+      }
+      presenceData.details = "Learning:";
+      presenceData.state = camps[page];
+    }
+    //presenceData.details = "Learning:";
+    //presenceData.state = "Writing Code!";
+  } else if (["/login", "/login/"].includes(page)) {
+    presenceData.details = "Signing in";
+  } else if (page.startsWith("/news/")) {
+    presenceData.details = "Reading the news";
   }
   presence.setActivity(presenceData);
 });
Index: websites/H/Hdfilmcehennemi/presence.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/websites/H/Hdfilmcehennemi/presence.ts b/websites/H/Hdfilmcehennemi/presence.ts
new file mode 100644
--- /dev/null	(date 1614933496051)
+++ b/websites/H/Hdfilmcehennemi/presence.ts	(date 1614933496051)
@@ -0,0 +1,65 @@
+const presence = new Presence({
+    clientId: "817292272508731392"
+  })
+
+const pages: {[index: string]:any} = {
+  "/film-istekleri-yap/": "Film İstekleri",
+  "/site-hakkinda/": "Yasal Bildirim",
+  "/iletisim/": "İletişim"
+}
+
+const categories: {[index: string]:any} = {
+  "/imdb-7-puan-uzeri-filmler/": "Imdb 7+ Filmler",
+  "/en-cok-yorumlananlar/": "En Çok Yorumlananlar",
+  "/en-cok-begenilen-filmler-izle/": "En Çok Beğenilenler",
+  "/dil/turkce-dublajli-film-izleyin/": "Türkçe Dublaj",
+  "/dil/turkce-altyazili-film-izleme-sitesi/": "Türkçe Altyazı"
+}
+
+function myOutsideHeavyLiftingFunction(){
+
+}
+
+setInterval(myOutsideHeavyLiftingFunction, 10000);
+presence.on("UpdateData", async () => {
+
+  let page = document.location.pathname;
+  let urlParams = new URLSearchParams(window.location.search);
+  let myParam = urlParams.get('s');
+
+  if (!page.endsWith("/")) page = page + "/";
+
+  const presenceData: PresenceData = {
+    largeImageKey: "large"
+  };
+
+  if (pages[page]) {
+    presenceData.details = "Sayfada:";
+    presenceData.state = pages[page];
+  } else if (categories[page]) {
+    presenceData.details = "Film Arıyor:";
+    presenceData.state = categories[page];
+  } else if (page.includes("/category/") || page.includes("/tur/") || page.includes("/yil/")) {
+    presenceData.details = "Film Arıyor:";
+    // @ts-ignore
+    let category = document.getElementsByClassName("tab-title")[0].firstChild.innerHTML;
+    category.includes("Kategorisi") ? category = category.replace(/Kategorisi(.*)/, "") : "";
+    category.includes("izle") ? category = category.replace(/izle(.*)/, "") : "";
+    presenceData.details = "Geziniyor:"
+    presenceData.state = category
+  } else if (document.getElementById("player")) {
+    let movieName = document.getElementsByClassName("movie-title")[0].children[1].innerHTML;
+    movieName.includes("izle") ? movieName = movieName.replace(/izle(.*)/, "") : "";
+    presenceData.details = "Film İzliyor:";
+    presenceData.state = movieName;
+    presenceData.
+  } else if (myParam) {
+    presenceData.details = "Film Arıyor:";
+    presenceData.state = myParam;
+  } else {
+    presenceData.details = "Geziniyor";
+  }
+
+
+  presence.setActivity(presenceData);
+});
Index: websites/H/Hdfilmcehennemi/tsconfig.json
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/websites/H/Hdfilmcehennemi/tsconfig.json b/websites/H/Hdfilmcehennemi/tsconfig.json
new file mode 100644
--- /dev/null	(date 1614894035479)
+++ b/websites/H/Hdfilmcehennemi/tsconfig.json	(date 1614894035479)
@@ -0,0 +1,6 @@
+{
+  "extends": "../../../tsconfig.json",
+  "compilerOptions": {
+    "outDir": "./dist/"
+  }
+}
Index: websites/H/Hdfilmcehennemi/dist/metadata.json
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/websites/H/Hdfilmcehennemi/dist/metadata.json b/websites/H/Hdfilmcehennemi/dist/metadata.json
new file mode 100644
--- /dev/null	(date 1614894011841)
+++ b/websites/H/Hdfilmcehennemi/dist/metadata.json	(date 1614894011841)
@@ -0,0 +1,27 @@
+{
+  "$schema": "https://schemas.premid.app/metadata/1.3",
+  "service": "Hdfilmcehennemi",
+  "author": {
+    "name": "RedS",
+    "id": "389383409430167562"
+  },
+  "category": "videos",
+  "logo": "https://i.imgur.com/XGbtqVa.png",
+  "thumbnail": "https://i.imgur.com/VmH80Em.png",
+  "version": "1.0.0",
+  "color": "#000000",
+  "tags": [
+    "movie",
+    "turkish",
+    "video",
+    "hdfilmcehennemi"
+  ],
+  "url": [
+    "www.hdfilmcehennemi.net",
+    "hdfilmcehennemi.net"
+  ],
+  "description": {
+    "tr": "Hdfilmcehennemi - Türkiye'nin en hızlı hd film izleme sitesi. Tek ve gerçek hdfilmcehennemi sitesi.",
+    "en": "Hdfilmcehennemi is a Turkish service that provides latest movies."
+  }
+}
